src/rpc/rpc.h

```diff
@@ -97,6 +108,38 @@ enum RPCStatus {
   kRPCTimeOut,
 };
 
+/*!
+ * \brief Serialize one meta of RPCMessage to data buffer.
+ *
+ * The serialized data format is:
+ * 
+ * |service_id| -> int32_t
+ * |msg_seq   | -> int64_t
+ * |client_id | -> int32_t
+ * |server_id | -> int32_t
+ * |data_size | -> int64_t
+ * |data      | -> string
+ * |has_tensor| -> int8_t 1 for true and 0 for false
+ *
+ * Note: The real data of tensor will not be serialized to buffer, and
+ * the the memory management of data buffer should be done by user.
+ *
+ * \param msg RPC message to serilaize
+ * \param size the total size of serialized data
+ * \return data buffer
+ */
+char* SerializeRPCMeta(const RPCMessage& msg, int64_t* size);
+
+/*!
+ * \brief Deserialize one meta of RPCMessage from data buffer
+ *
+ * \param msg RPC message to deserilaize
+ * \param buffer buffer holding the serialized data
+ * \param size buffer size
+ * \return has_tensor
+ */
+bool DeserializeRPCMeta(RPCMessage* msg, char* buffer, int64_t size);
```

Comment:
You should use the `Write` and `Read` interfaces.

Reply:
@zheng-da @jermainewang Please see the new code